src/NuGet.Core/NuGet.Commands/PackCommandRunner.cs

```diff
@@ -186,7 +186,11 @@ private PackageArchiveReader BuildPackage(PackageBuilder builder, string outputP
 
             if (!string.IsNullOrEmpty(_packArgs.Suffix))
             {
-                builder.Version = new NuGetVersion(builder.Version.Version, _packArgs.Suffix);
+                builder.Version = new NuGetVersion($"{builder.Version.ToNormalizedString()}-{_packArgs.Suffix}");
```

Comment:
Seems like this can output two suffixes if _packArgs.Verson already has a suffix